### ISSUE TITLE
railsContext enhancement

### ISSUE THREAD

**Issue by Tamiyadd** [shakacode/react_on_rails#367]
Hi @justin808 ,
good job with railsContext, it's a very great feature and i'm going to use it a lot!

I'm thinking about a little enhancement about it.

Don't you think is better to change the `host` parameter in the default railsContext values

```
host: uri.host, # foo.com
```

to 

```
host: request.host_with_port, # foo.com
```

so in development and in some production configuration you will have the full host with port? 
or maybe simply add another parameter to the default values that eventually contains port?

what do you think?

**Comment by justin808** [comment]
Does this have it: `href: request.original_url,`

In any case, let's add the `port` number as a value: https://developer.mozilla.org/en-US/docs/Web/API/Location

```
Location.port
Is a DOMString containing the port number of the URL.
```

I'm going to also add a flag that let's the JavaScript code know if we're server rendering.

```
serverRendering: boolean
```

True if server rendering, false if client rendering.

**Comment by justin808** [comment]
Fixed in 5.1.1.
